### ISSUE TITLE
URLs without a port number after : in authority section may not be invalid

### ISSUE THREAD

**Issue by tawAsh1** [line/armeria#4932]
In Armeria's Endpoint implementation, URLs without a port number after the : are considered [invalid](https://github.com/line/armeria/blob/9a8c0a05f6426992b0ed5298d52b6a19e312ea93/core/src/main/java/com/linecorp/armeria/client/Endpoint.java#L111).
For example, the following URL `http://example.com:/hogehoge`

This URL looks strange at first glance, but according to the RFC, it is defined as **valid** in the ABNF.

[RFC3986 Appendix A](https://datatracker.ietf.org/doc/html/rfc3986#appendix-A)

```
authority = [ userinfo "@" ] host [ ":" port ]
userinfo = *( unreserved / pct-encoded / sub-delims / ":" )
host = IP-literal / IPv4address / reg-name
port = *DIGIT
```

According to this ABNF, *DIGIT for the port section.
Since * denotes zero or more elements, it is Valid even if there is no number in the port section.

I also investigated several URL (URL) Validation rules and found the following that are judged as Valid

- https://owasp.org/www-community/OWASP_Validation_Regex_Repository
- java.net.URI
- org.apache.commons.validator.routines.UrlValidator

Perhaps URLs like these should be accepted as valid.

**Comment by jrhee17** [comment]
I'm not sure if this matches your use-case, but there is a similar known issue #4577 
Does this match your issue?

If it does, I think we agree but we just weren't able to prioritize it 😅

**Comment by ikhoon** [comment]
I didn't know that `example.com:` is a valid pattern. We may silently ignore `:` and normalize it to `example.com`.

**Comment by ikhoon** [comment]
We are about to release the new version. Please normalize the URL on your side before fixing the issue. 

Q: Are there cases where the `host:` pattern is used as an authority?

**Comment by minwoox** [comment]
Closed by #5561.
Please feel free to reopen if it's not addressed.

**Comment by Yang-33** [comment]
@Leewongi0731 @minwoox this was nice, thanks!
